pkg/util/bandwidth/utils.go

```diff
@@ -17,50 +17,109 @@ limitations under the License.
 package bandwidth
 
 import (
+	"errors"
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-var minRsrc = resource.MustParse("1k")
-var maxRsrc = resource.MustParse("1P")
+const (
+	directionIngress = "ingress"
+	directionEgress  = "egress"
+)
+
+var (
+	minBandwidth = resource.MustParse("1k")
+	maxBandwidth = resource.MustParse("1P")
+
+	minBurst = resource.MustParse("1k")
+	maxBurst = resource.MustParse("4G") // comply with bandwidth plugin limit
+)
 
 func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
-	if rsrc.Value() < minRsrc.Value() {
+	if rsrc.Value() < minBandwidth.Value() {
 		return fmt.Errorf("resource is unreasonably small (< 1kbit)")
 	}
-	if rsrc.Value() > maxRsrc.Value() {
+	if rsrc.Value() > maxBandwidth.Value() {
 		return fmt.Errorf("resoruce is unreasonably large (> 1Pbit)")
 	}
 	return nil
 }
 
-// ExtractPodBandwidthResources extracts the ingress and egress from the given pod annotations
-func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, egress *resource.Quantity, err error) {
+func validateBurstIsReasonable(rsrc *resource.Quantity) error {
+	if rsrc.Value() < minBurst.Value() {
+		return fmt.Errorf("resource is unreasonably small (< 1kbit)")
+	}
+	if rsrc.Value() > maxBurst.Value() {
+		return fmt.Errorf("resoruce is unreasonably large (> 4GB)")
```

Comment:
nit, you carried on the typo :)

Reply:
haha, sorry about that, I've updated the error message to clarify the source type, please see the latest commit.